src/sleep/sleep.c

```diff
@@ -390,7 +390,15 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
                         }
                 }
 
-                /* Do not suspend more than HibernateDelaySec= */
+                /* Do not suspend more than HibernateDelaySec= unless HibernateOnACPower=no and currently on AC power */
+                if (!sleep_config->hibernate_on_ac_power) {
+                        /* Do not allow "decay" to suspend if the system has no battery. */
+                        if (hashmap_isempty(last_capacity))
+                                log_warning("HibernateOnACPower=no was ignored because the system does not have a battery.");
+                        else if (on_ac_power())
```

Comment:
Please check for error here

Reply:
Did you mean to just check > 0 instead of truthy? I just made that change.

I don't see any reason to log an error here, if that is what you're suggesting.